infra/modules/azure_cosmos_account/variables.tf

```diff
@@ -0,0 +1,148 @@
+# ------------ GENERAL ------------ #
+variable "tags" {
+  type        = map(any)
+  description = "Resources tags"
+}
+
+variable "environment" {
+  type = object({
+    prefix          = string
+    env_short       = string
+    location        = string
+    domain          = optional(string)
+    app_name        = string
+    instance_number = string
+  })
+
+  description = "Values which are used to generate resource names and location short names. They are all mandatory except for domain, which should not be used only in the case of a resource used by multiple domains."
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "Resource group to deploy resources to"
+}
+
+# ------------ COSMOS ------------ #
+variable "subnet_pep_id" {
+  type        = string
+  description = "Id of the subnet which holds private endpoints"
+}
+
+variable "private_dns_zone_resource_group_name" {
+  type        = string
+  description = "(Optional) The name of the resource group holding private DNS zone to use for private endpoints. Default is Virtual Network resource group"
+  default     = null
+}
+
+variable "primary_geo_location" {
+  type = object({
+    location       = optional(string, null)
+    zone_redundant = optional(bool, true)
+  })
+  description = "Primary geo location for Cosmos DB account. Set location if you want to deploy the cosmos account in a different region than the default."
+
+  default = {
+    location       = null
+    zone_redundant = true
+  }
+}
+
+variable "secondary_geo_locations" {
+  type = list(object({
+    location          = optional(string, null)
+    failover_priority = optional(number, null)
+    zone_redundant    = optional(bool, true)
+  }))
+  description = "(Optional) Secondary geo locations for Cosmos DB account. Failover priority determines the order in which regions will take over in case of a regional outage. If failover priority is not set, the items order is used."
+  default     = []
+}
+
+variable "customer_managed_key" {
+  type = object({
+    enabled                   = optional(bool, false)
+    user_assigned_identity_id = optional(string, null)
+    key_vault_key_id          = optional(string, null)
+  })
+  description = "(Optional) Customer managed key to use for encryption"
+  default     = { enabled = false }
+
+  validation {
+    condition = (
+      (!var.customer_managed_key.enabled) ||
+      (var.customer_managed_key.enabled && var.customer_managed_key.user_assigned_identity_id != null && var.customer_managed_key.key_vault_key_id != null)
+    )
+    error_message = "Either 'user_assigned_identity_id' or 'key_vault_key_id' must be provided when 'enabled' is set to true."
+  }
+}
+
+variable "force_public_network_access_enabled" {
+  type        = bool
+  description = "(Optional) Whether the Storage Account permits public network access or not. Defaults to false."
+  default     = false
+}
+
+variable "consistency_policy" {
```

Comment:
Is it necessary for this variable to be highly configurable? Perhaps we could identify a recurring preset that better fits our use cases

Reply:
Actually, after [Danilo's comment](https://github.com/pagopa/dx/pull/130#discussion_r1791742751) some presets have been implemented (Default, HighConsistency, HighPerformance, BalancedStaleness)

However, by setting the consistency_preset to "Custom", it is possible to configure the consistency settings in detail (but still not needed if you are not a cosmos ninja 😄 🥷 )